packages/@aws-cdk/core/lib/stack.ts

```diff
@@ -263,6 +280,10 @@ export class Stack extends Construct implements ITaggable {
    * Add a dependency between this stack and another stack
    */
   public addDependency(stack: Stack, reason?: string) {
+    if (process.env.CDK_DEBUG_DEPS) {
```

Comment:
Leaving this in? Should be done after `reason` has been given a default value.

Reply:
Yes, why not. Moved until after `reason` is set.